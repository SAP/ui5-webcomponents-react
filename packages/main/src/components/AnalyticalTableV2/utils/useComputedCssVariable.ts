import type { RefObject } from 'react';
import { useEffect, useState } from 'react';

export function useComputedCssVariable<T extends HTMLElement>(ref: RefObject<T>, variableName: string): number | null {
  const [value, setValue] = useState<number | null>(null);

  useEffect(() => {
    if (ref.current) {
      const computedStyles = getComputedStyle(ref.current);
      const variableValue = computedStyles.getPropertyValue(variableName).trim();

      // Attempt to parse the value as a number
      const parsedValue = parseFloat(variableValue);
      setValue(isNaN(parsedValue) ? null : parsedValue);
    }
  }, [variableName, ref]);

  return value;
}
