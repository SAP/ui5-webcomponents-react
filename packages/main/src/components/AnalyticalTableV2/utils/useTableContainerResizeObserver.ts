import type { RefObject } from 'react';
import { useState, useEffect } from 'react';

export const useTableContainerResizeObserver = (tableContainerRef: RefObject<HTMLDivElement>) => {
  const [tableWidth, setTableWidth] = useState(0);
  const [horizontalScrollbarHeight, setHorizontalScrollbarHeight] = useState(0);

  useEffect(() => {
    const tableContainer = tableContainerRef.current;

    const resizeObserver = new ResizeObserver((entries) => {
      for (const entry of entries) {
        if (entry.borderBoxSize) {
          const { borderBoxSize, contentBoxSize } = entry;
          const borderBoxHeight = borderBoxSize[0].blockSize;
          const contentBoxHeight = contentBoxSize[0].blockSize;
          if (borderBoxHeight > contentBoxHeight) {
            setHorizontalScrollbarHeight(borderBoxHeight - contentBoxHeight);
          } else {
            setHorizontalScrollbarHeight(0);
          }

          const borderBoxWidth = borderBoxSize[0].inlineSize;
          setTableWidth(borderBoxWidth);
        }
      }
    });

    if (tableContainer) {
      resizeObserver.observe(tableContainer, { box: 'border-box' });
    }

    return () => {
      if (tableContainer) {
        resizeObserver.unobserve(tableContainer);
      }
    };
  }, [tableContainerRef]);

  return { tableWidth, horizontalScrollbarHeight };
};
