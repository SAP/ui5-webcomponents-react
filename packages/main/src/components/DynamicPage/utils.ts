export const DynamicPageCssVariables = {
  headerDisplay: '--_ui5wcr_DynamicPage_header_display',
  titleFontSize: '--_ui5wcr_DynamicPage_title_fontsize'
};
