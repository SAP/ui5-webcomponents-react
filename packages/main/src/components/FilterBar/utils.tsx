import { cloneElement } from 'react';

const inputTagNames = new Set([
  'UI5-COMBOBOX',
  'UI5-DATE-PICKER',
  'UI5-DATERANGE-PICKER',
  'UI5-DATETIME-PICKER',
  'UI5-DURATION-PICKER',
  'UI5-INPUT',
  'UI5-MULTI-INPUT',
  'UI5-RATING-INDICATOR',
  'UI5-STEP-INPUT',
  'UI5-TIME-PICKER'
]);

export const filterValue = (ref, child) => {
  const tagName = ref.tagName;
  let filterItemProps = {};
  if (inputTagNames.has(tagName)) {
    filterItemProps = { value: ref.value };
  }
  if (tagName === 'UI5-SELECT' || tagName === 'UI5-MULTI-COMBOBOX') {
    const selectedIndices = Array.from(ref.children as HTMLCollectionOf<any>)
      .map((item, index) => (item.selected ? index : false))
      .filter((el) => el !== false);
    const selectedIndicesSet = new Set(selectedIndices);
    const options = child.props.children.props.children?.map((item, index) => {
      if (selectedIndicesSet.has(index)) {
        return cloneElement(item, { selected: true });
      }
      return cloneElement(item, { selected: false });
    });
    filterItemProps = { children: options };
  }
  if (tagName === 'UI5-SWITCH' || tagName === 'UI5-CHECKBOX') {
    filterItemProps = { checked: ref.checked };
  }
  return filterItemProps;
};

export const renderSearchWithValue = (renderSearchElement, searchValue) => {
  return cloneElement(renderSearchElement, { value: searchValue ?? renderSearchElement?.props?.value });
};
