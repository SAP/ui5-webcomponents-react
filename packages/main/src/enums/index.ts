// This is an autogenerated file, please do not modify this file manually.
// In case you added a new file to the /enums folder, pleas rerun the scripts/create-enum-export.js script.

export * from './AnalyticalTableScrollMode';
export * from './AvatarColorScheme';
export * from './AvatarGroupType';
export * from './AvatarShape';
export * from './AvatarSize';
export * from './BarDesign';
export * from './BreadcrumbsDesign';
export * from './BreadcrumbsSeparatorStyle';
export * from './BusyIndicatorSize';
export * from './ButtonDesign';
export * from './CalendarSelection';
export * from './CalendarSelectionMode';
export * from './CalendarType';
export * from './CarouselArrowsPlacement';
export * from './ContentDensity';
export * from './DeviationIndicator';
export * from './FCLLayout';
export * from './FlexBoxAlignItems';
export * from './FlexBoxDirection';
export * from './FlexBoxJustifyContent';
export * from './FlexBoxWrap';
export * from './GlobalStyleClasses';
export * from './GridPosition';
export * from './IllustrationMessageType';
export * from './IndicationColor';
export * from './InputType';
export * from './LinkDesign';
export * from './ListGrowingMode';
export * from './ListItemType';
export * from './ListMode';
export * from './ListSeparators';
export * from './LoaderType';
export * from './MediaGalleryItemLayout';
export * from './MediaGalleryLayout';
export * from './MediaGalleryMenuHorizontalAlign';
export * from './MediaGalleryMenuVerticalAlign';
export * from './MessageBoxActions';
export * from './MessageBoxTypes';
export * from './MessageStripDesign';
export * from './ObjectPageMode';
export * from './PageBackgroundDesign';
export * from './PanelAccessibleRole';
export * from './PopoverHorizontalAlign';
export * from './PopoverPlacementType';
export * from './PopoverVerticalAlign';
export * from './Priority';
export * from './SemanticColor';
export * from './SideContentFallDown';
export * from './SideContentPosition';
export * from './SideContentVisibility';
export * from './Size';
export * from './SwitchDesign';
export * from './TabLayout';
export * from './TableGrowingMode';
export * from './TableMode';
export * from './TableRowType';
export * from './TableScaleWidthMode';
export * from './TableSelectionBehavior';
export * from './TableSelectionMode';
export * from './TableVisibleRowCountMode';
export * from './TabsOverflowMode';
export * from './TextAlign';
export * from './Themes';
export * from './TimelineLayout';
export * from './TitleLevel';
export * from './ToastPlacement';
export * from './ToolbarDesign';
export * from './ToolbarStyle';
export * from './UploadState';
export * from './ValueColor';
export * from './ValueState';
export * from './VerticalAlign';
export * from './WrappingType';
