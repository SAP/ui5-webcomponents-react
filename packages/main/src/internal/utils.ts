import { Children, Fragment, ReactNode } from 'react';

export function flattenFragments(children: ReactNode | ReactNode[], depth = 1): ReactNode[] {
  const flatChildren = [];
  if (Array.isArray(children)) {
    return children;
  }
  const removeFragments = (element, level = 0) => {
    if (!element) {
      return;
    }
    if (element.type === Fragment && level <= depth) {
      Children.toArray(element.props?.children).forEach((item) => {
        removeFragments(item, level + 1);
      });
    } else {
      flatChildren.push(element);
    }
  };
  removeFragments(children);

  return flatChildren;
}
