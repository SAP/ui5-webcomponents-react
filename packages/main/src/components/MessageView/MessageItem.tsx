import { CssSizeVariables } from '@ui5/webcomponents-react-base/dist/CssSizeVariables';
import { StyleClassHelper } from '@ui5/webcomponents-react-base/dist/StyleClassHelper';
import { ThemingParameters } from '@ui5/webcomponents-react-base/dist/ThemingParameters';
import { CustomListItem } from '@ui5/webcomponents-react/dist/CustomListItem';
import { FlexBox } from '@ui5/webcomponents-react/dist/FlexBox';
import { FlexBoxAlignItems } from '@ui5/webcomponents-react/dist/FlexBoxAlignItems';
import { FlexBoxDirection } from '@ui5/webcomponents-react/dist/FlexBoxDirection';
import { Icon } from '@ui5/webcomponents-react/dist/Icon';
import { Label } from '@ui5/webcomponents-react/dist/Label';
import { ListItemType } from '@ui5/webcomponents-react/dist/ListItemType';
import { MessageViewContext } from '@ui5/webcomponents-react/dist/MessageViewContext';
import { ValueState } from '@ui5/webcomponents-react/dist/ValueState';
import { CommonProps } from '@ui5/webcomponents-react/interfaces/CommonProps';
import { Ui5DomRef } from '@ui5/webcomponents-react/interfaces/Ui5DomRef';
import React, { forwardRef, ReactNode, ReactNodeArray, Ref, useContext } from 'react';
import { createUseStyles } from 'react-jss';
import { getIconNameForType } from './utils';

export interface MessageItemPropTypes extends CommonProps {
  /**
   * Specifies the title of the message
   */
  titleText: string;

  /**
   * Specifies the subtitle of the message
   */
  subtitleText?: string;

  /**
   * Defines the number of messages for a given message.
   */
  counter?: number;

  /**
   * Specifies the type of the message
   */
  type: ValueState | keyof typeof ValueState;

  /**
   * Name of a message group the current item belongs to.
   */
  groupName?: string;

  /**
   * Specifies detailed description of the message
   */
  children: ReactNode | ReactNodeArray;
}

const useStyles = createUseStyles(
  {
    listItem: {
      height: CssSizeVariables.sapWcrMessageViewListItemHeightSingle
    },
    message: {
      padding: '0.25rem 0',
      width: '100%',
      maxWidth: '100%',
      overflow: 'hidden',
      paddingRight: '1rem',
      boxSizing: 'border-box'
    },
    withSubtitle: {
      height: CssSizeVariables.sapWcrMessageViewListItemHeightByLine
    },
    withChildren: {
      paddingRight: '0rem'
    },
    iconContainer: {
      width: '3rem',
      minWidth: '3rem',
      height: '2.25rem',
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center'
    },
    icon: {
      width: '1rem',
      height: '1rem'
    },
    title: {
      fontFamily: ThemingParameters.sapFontHeaderFamily,
      fontSize: ThemingParameters.sapFontHeader6Size,
      color: ThemingParameters.sapGroup_TitleTextColor,
      maxWidth: '100%',
      whiteSpace: 'nowrap',
      overflow: 'hidden',
      textOverflow: 'ellipsis',
      flex: '1 1 auto',
      '& + $subtitle': {
        marginTop: '0.25rem',
        cursor: 'inherit'
      }
    },
    subtitle: {},
    counter: {
      color: ThemingParameters.sapContent_MarkerTextColor,
      fontFamily: ThemingParameters.sapFontFamily,
      fontSize: ThemingParameters.sapFontSize,
      paddingLeft: '1rem',
      flex: 'none'
    },
    navigation: {
      height: '0.875rem',
      width: '0.875rem',
      padding: '0 0.9375rem',
      flexShrink: 0
    },
    typeError: {
      '& $icon': {
        color: ThemingParameters.sapNegativeElementColor
      }
    },
    typeSuccess: {
      '& $icon': {
        color: ThemingParameters.sapPositiveElementColor
      }
    },
    typeWarning: {
      ' & $icon': {
        color: ThemingParameters.sapCriticalElementColor
      }
    },
    typeInformation: {
      '& $icon': {
        // color: ThemingParameters.sapInformativeElementColor
        color: ThemingParameters.sapNeutralElementColor
      }
    },
    typeNone: {
      '& $icon': {
        color: ThemingParameters.sapNeutralElementColor
      }
    }
  },
  { name: 'MessageItem' }
);

const MessageItem = forwardRef((props: MessageItemPropTypes, ref: Ref<Ui5DomRef>) => {
  const { titleText, subtitleText, counter, type, children, className, ...rest } = props;

  const { selectMessage } = useContext(MessageViewContext);

  const classes = useStyles();

  const listItemClasses = StyleClassHelper.of(classes.listItem, Reflect.get(classes, `type${type}`)).putIfPresent(
    className
  );
  if (subtitleText) {
    listItemClasses.put(classes.withSubtitle);
  }

  const messageClasses = StyleClassHelper.of(classes.message);
  if (children) {
    messageClasses.put(classes.withChildren);
  }

  const handleListItemClick = (e) => {
    if (children) {
      selectMessage(props);
      if (typeof rest.onClick === 'function') {
        rest.onClick(e);
      }
    }
  };
  return (
    <CustomListItem
      onClick={handleListItemClick}
      data-title={titleText}
      data-type={type}
      type={children ? ListItemType.Active : ListItemType.Inactive}
      {...rest}
      className={listItemClasses.className}
      ref={ref}
    >
      <FlexBox alignItems={FlexBoxAlignItems.Center} className={messageClasses.className}>
        <div className={classes.iconContainer}>
          <Icon name={getIconNameForType(type as ValueState)} className={classes.icon} />
        </div>
        <FlexBox
          direction={FlexBoxDirection.Column}
          style={{ flex: 'auto', whiteSpace: 'nowrap', overflow: 'hidden', textOverflow: 'ellipsis' }}
        >
          {titleText && <span className={classes.title}>{titleText}</span>}
          {subtitleText && <Label className={classes.subtitle}>{subtitleText}</Label>}
        </FlexBox>
        {counter != null && <span className={classes.counter}>{counter}</span>}
        {children && <Icon className={classes.navigation} name="slim-arrow-right" />}
      </FlexBox>
    </CustomListItem>
  );
});

MessageItem.displayName = 'MessageItem';

MessageItem.defaultProps = {
  type: ValueState.None
};

export { MessageItem };
