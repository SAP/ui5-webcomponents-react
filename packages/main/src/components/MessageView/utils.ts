import iconAlert from '@ui5/webcomponents-icons/dist/alert.js';
import iconError from '@ui5/webcomponents-icons/dist/error.js';
import iconInformation from '@ui5/webcomponents-icons/dist/information.js';
import iconSysEnter from '@ui5/webcomponents-icons/dist/sys-enter-2.js';
import { ValueState } from '../../enums/ValueState';

export const getIconNameForType = (type: ValueState | keyof typeof ValueState): string => {
  switch (type) {
    case ValueState.Error:
    case 'Error':
      return iconError;
    case ValueState.Success:
    case 'Success':
      return iconSysEnter;
    case ValueState.Warning:
    case 'Warning':
      return iconAlert;
    default:
      return iconInformation;
  }
};
