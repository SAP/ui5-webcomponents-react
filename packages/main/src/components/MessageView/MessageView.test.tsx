import { fireEvent, render, screen } from '@shared/tests';
import { createCustomPropsTest } from '@shared/tests/utils';
import { MessageView, MessageViewDomRef } from './index';
import { MessageItem } from './MessageItem';
import React, { createRef, forwardRef } from 'react';
import { ValueState } from '../../enums/ValueState';

const TestComponent = forwardRef((props, ref) => {
  return (
    <MessageView {...props} ref={ref}>
      <MessageItem titleText="Error" type={ValueState.Error}>
        Error
      </MessageItem>
      <MessageItem titleText="Warning" type={ValueState.Warning}>
        Warning
      </MessageItem>
      <MessageItem titleText="Success" type={ValueState.Success}>
        Success
      </MessageItem>
      <MessageItem titleText="Information" type={ValueState.Information}>
        Information
      </MessageItem>
      <MessageItem titleText="None" type={ValueState.None}>
        None
      </MessageItem>
    </MessageView>
  );
});

describe('MessageView', () => {
  it('renders list view', () => {
    const { asFragment } = render(<TestComponent />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders list view with groups', () => {
    const { asFragment } = render(<TestComponent groupItems />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render no filter when only one message type is used', () => {
    render(
      <MessageView>
        <MessageItem titleText="Error" type={ValueState.Error}>
          Error
        </MessageItem>
      </MessageView>
    );

    expect(screen.queryAllByText('All')).toHaveLength(0);
  });

  it('navigate to details and back to list', () => {
    const ref = createRef<MessageViewDomRef>();
    render(<TestComponent ref={ref} />);

    fireEvent.click(screen.getByText('Error').parentElement.parentElement.parentElement);

    expect(screen.queryAllByText('Error')).toHaveLength(3); // list, details page header and children

    ref.current.navigateBack();

    expect(screen.queryAllByText('Error')).toHaveLength(1); // list
  });

  createCustomPropsTest(MessageView);
});
