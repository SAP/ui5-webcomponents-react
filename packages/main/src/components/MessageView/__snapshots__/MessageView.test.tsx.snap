// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageView renders list view 1`] = `
<DocumentFragment>
  <div
    class="MessageView-container"
  >
    <div>
      <ui5-bar
        design="Header"
        ui5-bar=""
      >
        <ui5-segmented-button
          slot="startContent"
          ui5-segmented-button=""
        >
          <ui5-segmented-button-item
            data-key="All"
            pressed="true"
            ui5-segmented-button-item=""
          >
            All
          </ui5-segmented-button-item>
          <ui5-segmented-button-item
            class="MessageView-button"
            data-key="Error"
            icon="error"
            ui5-segmented-button-item=""
          >
            1
          </ui5-segmented-button-item>
          <ui5-segmented-button-item
            class="MessageView-button"
            data-key="Warning"
            icon="alert"
            ui5-segmented-button-item=""
          >
            1
          </ui5-segmented-button-item>
          <ui5-segmented-button-item
            class="MessageView-button"
            data-key="Success"
            icon="sys-enter-2"
            ui5-segmented-button-item=""
          >
            1
          </ui5-segmented-button-item>
          <ui5-segmented-button-item
            class="MessageView-button"
            data-key="Information"
            icon="information"
            ui5-segmented-button-item=""
          >
            2
          </ui5-segmented-button-item>
        </ui5-segmented-button>
      </ui5-bar>
      <ui5-list
        accessible-role="list"
        busy-delay="1000"
        growing="None"
        mode="None"
        separators="All"
        ui5-list=""
      >
        <ui5-li-custom
          actionable=""
          class="MessageItem-listItem MessageItem-typeError"
          data-title="Error"
          data-type="Error"
          type="Active"
          ui5-li-custom=""
        >
          <div
            class="FlexBox-flexBox FlexBox-flexBoxDirectionRow FlexBox-justifyContentStart FlexBox-alignItemsCenter FlexBox-flexWrapNoWrap MessageItem-message MessageItem-withChildren"
          >
            <div
              class="MessageItem-iconContainer"
            >
              <ui5-icon
                class="MessageItem-icon"
                name="error"
                ui5-icon=""
              />
            </div>
            <div
              class="FlexBox-flexBox FlexBox-flexBoxDirectionColumn FlexBox-justifyContentStart FlexBox-alignItemsStretch FlexBox-flexWrapNoWrap"
              style="flex-basis: auto; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
            >
              <span
                class="MessageItem-title"
              >
                Error
              </span>
            </div>
            <ui5-icon
              class="MessageItem-navigation"
              name="slim-arrow-right"
              ui5-icon=""
            />
          </div>
        </ui5-li-custom>
        <ui5-li-custom
          actionable=""
          class="MessageItem-listItem MessageItem-typeWarning"
          data-title="Warning"
          data-type="Warning"
          type="Active"
          ui5-li-custom=""
        >
          <div
            class="FlexBox-flexBox FlexBox-flexBoxDirectionRow FlexBox-justifyContentStart FlexBox-alignItemsCenter FlexBox-flexWrapNoWrap MessageItem-message MessageItem-withChildren"
          >
            <div
              class="MessageItem-iconContainer"
            >
              <ui5-icon
                class="MessageItem-icon"
                name="alert"
                ui5-icon=""
              />
            </div>
            <div
              class="FlexBox-flexBox FlexBox-flexBoxDirectionColumn FlexBox-justifyContentStart FlexBox-alignItemsStretch FlexBox-flexWrapNoWrap"
              style="flex-basis: auto; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
            >
              <span
                class="MessageItem-title"
              >
                Warning
              </span>
            </div>
            <ui5-icon
              class="MessageItem-navigation"
              name="slim-arrow-right"
              ui5-icon=""
            />
          </div>
        </ui5-li-custom>
        <ui5-li-custom
          actionable=""
          class="MessageItem-listItem MessageItem-typeSuccess"
          data-title="Success"
          data-type="Success"
          type="Active"
          ui5-li-custom=""
        >
          <div
            class="FlexBox-flexBox FlexBox-flexBoxDirectionRow FlexBox-justifyContentStart FlexBox-alignItemsCenter FlexBox-flexWrapNoWrap MessageItem-message MessageItem-withChildren"
          >
            <div
              class="MessageItem-iconContainer"
            >
              <ui5-icon
                class="MessageItem-icon"
                name="sys-enter-2"
                ui5-icon=""
              />
            </div>
            <div
              class="FlexBox-flexBox FlexBox-flexBoxDirectionColumn FlexBox-justifyContentStart FlexBox-alignItemsStretch FlexBox-flexWrapNoWrap"
              style="flex-basis: auto; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
            >
              <span
                class="MessageItem-title"
              >
                Success
              </span>
            </div>
            <ui5-icon
              class="MessageItem-navigation"
              name="slim-arrow-right"
              ui5-icon=""
            />
          </div>
        </ui5-li-custom>
        <ui5-li-custom
          actionable=""
          class="MessageItem-listItem MessageItem-typeInformation"
          data-title="Information"
          data-type="Information"
          type="Active"
          ui5-li-custom=""
        >
          <div
            class="FlexBox-flexBox FlexBox-flexBoxDirectionRow FlexBox-justifyContentStart FlexBox-alignItemsCenter FlexBox-flexWrapNoWrap MessageItem-message MessageItem-withChildren"
          >
            <div
              class="MessageItem-iconContainer"
            >
              <ui5-icon
                class="MessageItem-icon"
                name="information"
                ui5-icon=""
              />
            </div>
            <div
              class="FlexBox-flexBox FlexBox-flexBoxDirectionColumn FlexBox-justifyContentStart FlexBox-alignItemsStretch FlexBox-flexWrapNoWrap"
              style="flex-basis: auto; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
            >
              <span
                class="MessageItem-title"
              >
                Information
              </span>
            </div>
            <ui5-icon
              class="MessageItem-navigation"
              name="slim-arrow-right"
              ui5-icon=""
            />
          </div>
        </ui5-li-custom>
        <ui5-li-custom
          actionable=""
          class="MessageItem-listItem MessageItem-typeNone"
          data-title="None"
          data-type="None"
          type="Active"
          ui5-li-custom=""
        >
          <div
            class="FlexBox-flexBox FlexBox-flexBoxDirectionRow FlexBox-justifyContentStart FlexBox-alignItemsCenter FlexBox-flexWrapNoWrap MessageItem-message MessageItem-withChildren"
          >
            <div
              class="MessageItem-iconContainer"
            >
              <ui5-icon
                class="MessageItem-icon"
                name="information"
                ui5-icon=""
              />
            </div>
            <div
              class="FlexBox-flexBox FlexBox-flexBoxDirectionColumn FlexBox-justifyContentStart FlexBox-alignItemsStretch FlexBox-flexWrapNoWrap"
              style="flex-basis: auto; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
            >
              <span
                class="MessageItem-title"
              >
                None
              </span>
            </div>
            <ui5-icon
              class="MessageItem-navigation"
              name="slim-arrow-right"
              ui5-icon=""
            />
          </div>
        </ui5-li-custom>
      </ui5-list>
    </div>
    <div />
  </div>
</DocumentFragment>
`;

exports[`MessageView renders list view with groups 1`] = `
<DocumentFragment>
  <div
    class="MessageView-container"
  >
    <div>
      <ui5-bar
        design="Header"
        ui5-bar=""
      >
        <ui5-segmented-button
          slot="startContent"
          ui5-segmented-button=""
        >
          <ui5-segmented-button-item
            data-key="All"
            pressed="true"
            ui5-segmented-button-item=""
          >
            All
          </ui5-segmented-button-item>
          <ui5-segmented-button-item
            class="MessageView-button"
            data-key="Error"
            icon="error"
            ui5-segmented-button-item=""
          >
            1
          </ui5-segmented-button-item>
          <ui5-segmented-button-item
            class="MessageView-button"
            data-key="Warning"
            icon="alert"
            ui5-segmented-button-item=""
          >
            1
          </ui5-segmented-button-item>
          <ui5-segmented-button-item
            class="MessageView-button"
            data-key="Success"
            icon="sys-enter-2"
            ui5-segmented-button-item=""
          >
            1
          </ui5-segmented-button-item>
          <ui5-segmented-button-item
            class="MessageView-button"
            data-key="Information"
            icon="information"
            ui5-segmented-button-item=""
          >
            2
          </ui5-segmented-button-item>
        </ui5-segmented-button>
      </ui5-bar>
      <ui5-list
        accessible-role="list"
        busy-delay="1000"
        growing="None"
        mode="None"
        separators="All"
        ui5-list=""
      >
        <ui5-li-custom
          actionable=""
          class="MessageItem-listItem MessageItem-typeError"
          data-title="Error"
          data-type="Error"
          type="Active"
          ui5-li-custom=""
        >
          <div
            class="FlexBox-flexBox FlexBox-flexBoxDirectionRow FlexBox-justifyContentStart FlexBox-alignItemsCenter FlexBox-flexWrapNoWrap MessageItem-message MessageItem-withChildren"
          >
            <div
              class="MessageItem-iconContainer"
            >
              <ui5-icon
                class="MessageItem-icon"
                name="error"
                ui5-icon=""
              />
            </div>
            <div
              class="FlexBox-flexBox FlexBox-flexBoxDirectionColumn FlexBox-justifyContentStart FlexBox-alignItemsStretch FlexBox-flexWrapNoWrap"
              style="flex-basis: auto; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
            >
              <span
                class="MessageItem-title"
              >
                Error
              </span>
            </div>
            <ui5-icon
              class="MessageItem-navigation"
              name="slim-arrow-right"
              ui5-icon=""
            />
          </div>
        </ui5-li-custom>
        <ui5-li-custom
          actionable=""
          class="MessageItem-listItem MessageItem-typeWarning"
          data-title="Warning"
          data-type="Warning"
          type="Active"
          ui5-li-custom=""
        >
          <div
            class="FlexBox-flexBox FlexBox-flexBoxDirectionRow FlexBox-justifyContentStart FlexBox-alignItemsCenter FlexBox-flexWrapNoWrap MessageItem-message MessageItem-withChildren"
          >
            <div
              class="MessageItem-iconContainer"
            >
              <ui5-icon
                class="MessageItem-icon"
                name="alert"
                ui5-icon=""
              />
            </div>
            <div
              class="FlexBox-flexBox FlexBox-flexBoxDirectionColumn FlexBox-justifyContentStart FlexBox-alignItemsStretch FlexBox-flexWrapNoWrap"
              style="flex-basis: auto; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
            >
              <span
                class="MessageItem-title"
              >
                Warning
              </span>
            </div>
            <ui5-icon
              class="MessageItem-navigation"
              name="slim-arrow-right"
              ui5-icon=""
            />
          </div>
        </ui5-li-custom>
        <ui5-li-custom
          actionable=""
          class="MessageItem-listItem MessageItem-typeSuccess"
          data-title="Success"
          data-type="Success"
          type="Active"
          ui5-li-custom=""
        >
          <div
            class="FlexBox-flexBox FlexBox-flexBoxDirectionRow FlexBox-justifyContentStart FlexBox-alignItemsCenter FlexBox-flexWrapNoWrap MessageItem-message MessageItem-withChildren"
          >
            <div
              class="MessageItem-iconContainer"
            >
              <ui5-icon
                class="MessageItem-icon"
                name="sys-enter-2"
                ui5-icon=""
              />
            </div>
            <div
              class="FlexBox-flexBox FlexBox-flexBoxDirectionColumn FlexBox-justifyContentStart FlexBox-alignItemsStretch FlexBox-flexWrapNoWrap"
              style="flex-basis: auto; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
            >
              <span
                class="MessageItem-title"
              >
                Success
              </span>
            </div>
            <ui5-icon
              class="MessageItem-navigation"
              name="slim-arrow-right"
              ui5-icon=""
            />
          </div>
        </ui5-li-custom>
        <ui5-li-custom
          actionable=""
          class="MessageItem-listItem MessageItem-typeInformation"
          data-title="Information"
          data-type="Information"
          type="Active"
          ui5-li-custom=""
        >
          <div
            class="FlexBox-flexBox FlexBox-flexBoxDirectionRow FlexBox-justifyContentStart FlexBox-alignItemsCenter FlexBox-flexWrapNoWrap MessageItem-message MessageItem-withChildren"
          >
            <div
              class="MessageItem-iconContainer"
            >
              <ui5-icon
                class="MessageItem-icon"
                name="information"
                ui5-icon=""
              />
            </div>
            <div
              class="FlexBox-flexBox FlexBox-flexBoxDirectionColumn FlexBox-justifyContentStart FlexBox-alignItemsStretch FlexBox-flexWrapNoWrap"
              style="flex-basis: auto; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
            >
              <span
                class="MessageItem-title"
              >
                Information
              </span>
            </div>
            <ui5-icon
              class="MessageItem-navigation"
              name="slim-arrow-right"
              ui5-icon=""
            />
          </div>
        </ui5-li-custom>
        <ui5-li-custom
          actionable=""
          class="MessageItem-listItem MessageItem-typeNone"
          data-title="None"
          data-type="None"
          type="Active"
          ui5-li-custom=""
        >
          <div
            class="FlexBox-flexBox FlexBox-flexBoxDirectionRow FlexBox-justifyContentStart FlexBox-alignItemsCenter FlexBox-flexWrapNoWrap MessageItem-message MessageItem-withChildren"
          >
            <div
              class="MessageItem-iconContainer"
            >
              <ui5-icon
                class="MessageItem-icon"
                name="information"
                ui5-icon=""
              />
            </div>
            <div
              class="FlexBox-flexBox FlexBox-flexBoxDirectionColumn FlexBox-justifyContentStart FlexBox-alignItemsStretch FlexBox-flexWrapNoWrap"
              style="flex-basis: auto; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
            >
              <span
                class="MessageItem-title"
              >
                None
              </span>
            </div>
            <ui5-icon
              class="MessageItem-navigation"
              name="slim-arrow-right"
              ui5-icon=""
            />
          </div>
        </ui5-li-custom>
      </ui5-list>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
